### PR TITLE
Use appstreamcli to validate appstream

### PR DESCRIPTION
**Pull request by barthalion**

### CODE REVIEW

File: flatpak_builder_lint/appstream.py [hunk 1]
```diff
@@ -12,8 +12,33 @@
 def validate(path: str) -> dict:
     if not os.path.isfile(path):
         raise FileNotFoundError("AppStream file not found")
+
+    overrides = {
+        "all-categories-ignored": "error",
```
Comment (bbhtt) [inline flatpak_builder_lint/appstream.py:17]: This one doesn't exist on a release yet https://github.com/ximion/appstream/commit/0653448a469e3b61b4b5b54ca13eca7a92b65dd1

File: flatpak_builder_lint/appstream.py [hunk 2]
```diff
@@ -12,8 +12,33 @@
...
+        "cid-rnds-contains-hyphen": "error",
+        "content-rating-missing": "error",
+        "desktop-app-launchable-omitted": "error",
+        "desktop-file-not-found": "error",
+        "developer-id-missing": "error",
+        "invalid-child-tag-name": "error",
+        "metainfo-filename-cid-mismatch": "error",
+        "metainfo-legacy-path": "error",
+        "metainfo-multiple-components": "error",
+        "name-has-dot-suffix": "error",
```
Comment (bbhtt) [inline flatpak_builder_lint/appstream.py:32]: The summary one is also desired right https://github.com/flathub/flathub/issues/4635 ?

File: flatpak_builder_lint/appstream.py [hunk 3]
```diff
@@ -12,8 +12,33 @@
...
         raise FileNotFoundError("AppStream file not found")
+
+    overrides = {
+        "all-categories-ignored": "error",
+        "category-invalid": "error",
+        "cid-desktopapp-is-not-rdns": "error",
+        "cid-has-number-prefix": "error",
+        "cid-maybe-not-rdns": "error",
+        "cid-missing-affiliation-gnome": "error",
+        "cid-rnds-contains-hyphen": "error",
```
Comment (bbhtt) [inline flatpak_builder_lint/appstream.py:23]: Does this exist? I don't see it anywhere https://github.com/search?q=repo%3Aximion%2Fappstream+cid-rnds-contains-hyphen&type=code

Comment (bbhtt) [inline flatpak_builder_lint/appstream.py:23]:
```suggestion
        "cid-rdns-contains-hyphen": "error",
```

Ah

Comment (bbhtt) [reply flatpak_builder_lint/appstream.py:23]: This requires at least 1.0.0 https://github.com/ximion/appstream/commit/da92e3c9a9b5c6e9598afedb199744eee1a6a4eb

File: flatpak_builder_lint/appstream.py (hunk 1 above)
Comment (barthalion) [reply flatpak_builder_lint/appstream.py:17]: Good catch, we can backport it.

File: flatpak_builder_lint/appstream.py (hunk 2 above)
Comment (barthalion) [reply flatpak_builder_lint/appstream.py:32]: Not super sure. Dot suffix in app name looks immediately wrong, a dot in summary shouldn't cause build errors imho.

File: flatpak_builder_lint/checks/metainfo.py [hunk 4]
```diff
@@ -41,10 +39,11 @@ def _validate(self, path: str, appid: str) -> None:
...
             self.errors.add("appstream-failed-validation")
-        for err in metainfo_validation["stderr"].split(":", 1)[1:]:
-            self.appstream.add(err.strip())
-        for out in metainfo_validation["stdout"].splitlines()[1:]:
-            self.appstream.add(re.sub("^\u2022", "", out).strip())
+
+            for err in metainfo_validation["stderr"].split(":", 1)[1:]:
+                self.appstream.add(err.strip())
+            for out in metainfo_validation["stdout"].splitlines()[1:]:
+                self.appstream.add(re.sub("^\u2022", "", out).strip())
```
Comment (bbhtt) [inline flatpak_builder_lint/checks/metainfo.py:46]:
```suggestion
            for err in metainfo_validation["stderr"].splitlines():
                self.appstream.add(err.strip())
            for out in metainfo_validation["stdout"].splitlines()[:-1]:
                if out == "":
                    continue
                self.appstream.add(out.strip())
```

Comment (bbhtt) [reply flatpak_builder_lint/checks/metainfo.py:46]: Should make it look like this:

```
flatpak-builder-lint builddir builddir
{
    "errors": [
        "appstream-failed-validation"
    ],
    "appstream": [
        "E: ~:~: component-id-missing",
        "I: ~:10: developer-name-tag-deprecated"
    ],
    "message": "Please consult the documentation at https://docs.flathub.org/docs/for-app-authors/linter"
}
```

File: flatpak_builder_lint/appstream.py (hunk 2 above)
Comment (bbhtt) [reply flatpak_builder_lint/appstream.py:32]: Ok I guess I'll add it as a warning then, dots in summary are caught during review manually.

**Pull request merged**